src/Tasks/ResolveManifestFiles.cs

```diff
@@ -252,13 +256,27 @@ private static ITaskItem CreateAssemblyItem(ITaskItem item, string group, string
         }
 
         // Creates an output item for a file, with optional Group and IsData attributes.
-        private static ITaskItem CreateFileItem(ITaskItem item, string group, string targetPath, string includeHash, bool isDataFile)
+        private ITaskItem CreateFileItem(ITaskItem item, string group, string targetPath, string includeHash, bool isDataFile)
         {
             ITaskItem outputItem = new TaskItem(item.ItemSpec);
             item.CopyMetadataTo(outputItem);
             if (String.IsNullOrEmpty(targetPath))
             {
-                targetPath = GetItemTargetPath(outputItem);
+                targetPath = Path.GetFileName(item.ItemSpec);
+                //
+                // .NETCore Launcher.exe based deployment: If the file is apphost.exe, we need to set 'TargetPath' metadata
+                // to {assemblyname}.exe so that the file gets published as {assemblyname}.exe and not apphost.exe.
+                //
+                if (LauncherBasedDeployment && 
+                    targetPath.Equals(Constants.AppHostExe, StringComparison.InvariantCultureIgnoreCase) &&
+                    !String.IsNullOrEmpty(AssemblyName))
```

Comment:
I'm a little concerned about having this logic duplicated in several places. Could it be unified?

Reply:
The exactly identical check is in 2 locations so I didn't feel the need to factor it out. If you feel strongly about this, I can look into it.